src/MICore/UnixUtilities.cs

```diff
@@ -42,6 +42,12 @@ public static class UnixUtilities
             string pidFifo,
             string debuggerCmd)
         {
+            // On OSX, 'wait' will return once there is a status change from the launched process rather than for it to exit, so
```

Comment:
I _think_ any signal will cause it to return. But really I don't fully understand it - OSX has no documentation as to the behavior. It just returns immediately for OSX.